lib/puppet/indirector/rest.rb

```diff
@@ -111,6 +111,7 @@ def http_request(method, request, *args)
   end
 
   def find(request)
+    Puppet.deprecation_warning(_("Puppet::Indirector::Rest#find is deprecated. Use Puppet::HTTP::Client instead."))
```

Comment:
Would be good to verify this doesn't warn when `Puppet[:disable_warnings] = 'deprecations'`